src/scheduler/misc.c

```diff
@@ -889,43 +888,45 @@ is_num(char *str)
 	int i;
 	char c;
 	int colen_count = 0;
+	int str_len = -1;
 
 	if (str == NULL)
 		return -1;
 
 	if (str[0] == '-' || str[0] == '+')
 		str++;
 
-	for (i = 0; i < strlen(str) && (isdigit(str[i]) || str[i] == ':'); i++) {
+	str_len = strlen(str);
+	for (i = 0; i < str_len && (isdigit(str[i]) || str[i] == ':'); i++) {
 		if (str[i] == ':')
 			colen_count++;
 	}
 
 	/* is the string completly numeric or a time(HH:MM:SS or MM:SS) */
-	if (i == strlen(str) && colen_count <= 2)
+	if (i == str_len && colen_count <= 2)
 		return 1;
 
 	/* is the string a size */
-	if (i == strlen(str) - 2) {
+	if ((i == (str_len - 2)) || (i == (str_len - 1))) {
```

Comment:
The comment above this line is pointless. Please rephrase it or remove it.

All this test really tells you is that there is a character that is not a digit or a colon appearing somewhere in the string, not that it is formatted properly for a size. It doesn't distinguish between 123m and 12m3.

Reply:
"It doesn't distinguish between 123m and 12m3"
- Actually it does. "12m3" will satisfy the first if (c == 'k' || c == 'm' || c == 'g' || c == 't'). Then the next line c = tolower(str[i+1]); will move the pointer c from 'm' to '3', and then the inner if (c == 'b' || c == 'w' || c == '\0') will not be satisfied for "12m3". It would have ben satisfied if the string was "12mb"/"12mw"/"12m" instead.

The comment above is meant to convey that this check is testing for the string to be a size type resource like mem, I'll clarify that more.